byway/src/main/webapp/utils.js

```diff
@@ -0,0 +1,55 @@
+/** Script that contains shared functions, primarily for use in the third page */
+
+// object that communicates with the GMaps API service
+let directionsService;
+ 
+// object that renders display results on the map
+let directionsRenderer;
+
+// map object used in the route page
+let map; 
+
+// object that communicates with the Places API service
+let placesService;
+
+/**
+ * Uses placeId to retrieve details like coordinates, place name, etc.
+ * @param {String} placeId a textual identifier that uniquely identifies a place
+ * @return {Promise} result a Place object
+ */
+function findPlace(placeId) {
+  const request = {
+    placeId: placeId,
+    fields: ['name', 'geometry', 'place_id', 'photos', 'formatted_address']
+  }
+  const result = new Promise((resolve, reject) => {
+    placesService.getDetails(request, (result, status) => {
+      if(status == "OK") {
+        resolve(result);
+      } else {
+        alert("Status: " + status);
+        reject(result);
+      }
+    })
+  });
+  return result;
+}
+
+/** Initializes map on the page */
+function initMap() {
+  directionsService = new google.maps.DirectionsService();
+  directionsRenderer = new google.maps.DirectionsRenderer();
+ 
+  let mapOptions = {
+    zoom: 14,
+    // arbitrary center as it will get recentered to the route 
+    center: new google.maps.LatLng(0,0)
+  }
+  map = new google.maps.Map(document.getElementById('map'), mapOptions);
+  directionsRenderer.setMap(map);
+  placesService = new google.maps.places.PlacesService(map);
+  
+}
+
+/* export findPlace, initMap */
```

Comment:
Use exported instead of export for the eslint check

Reply:
I don't see the keyword "exported" in any javascript documentation.

Reply:
From the ESLint docs: [](https://eslint.org/docs/rules/no-unused-vars#exported)

Reply:
Oh it's an eslint specific thing, not js. Will change.